src/www/vpn_ipsec_pubkeys.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+/*
+ * Copyright (C) 2019 Pascal Mathis <mail@pascalmathis.com>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED ``AS IS'' AND ANY EXPRESS OR IMPLIED WARRANTIES,
+ * INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY
+ * AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY,
+ * OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+require_once("guiconfig.inc");
+require_once("filter.inc");
+require_once("services.inc");
+require_once("interfaces.inc");
+require_once("plugins.inc.d/ipsec.inc");
+require_once('phpseclib/Math/BigInteger.php');
+require_once('phpseclib/Crypt/Hash.php');
+require_once('phpseclib/Crypt/RSA.php');
+
+config_read_array('ipsec', 'publickey');
+ipsec_publickey_sort();
```

Comment:
Why? Wouldn’t it be better to write them in a sorted fashion?

Reply:
I wanted to avoid introducing too much complexity into the already fairly ugly IPsec implementation code (I have seen the ongoing discussions about this needing a rewrite one day) and oriented myself based on the IPsec Mobile Keys implementation, which also sorts in the same way - which is why I sticked to a similar way. I'm happy to take the different route of sorting keys during saving.